### ISSUE TITLE
[Request - Idea] Support font & icons colors theming

### ISSUE THREAD

**Issue by xerta555** [topjohnwu/Magisk#1945]
Hi @topjohnwu 

I have had this idea in my mind for some time now and I would like to know if it is possible at the moment:

Would it be possible to add an option to change the color of the Magisk Manager texts as well as the icons?

Perhaps this question could be more directed to @diareuse , who seems to know the Kotlin language very well.

Ty in advance.

**Comment by diareuse** [comment]
In the redesign we'll have in-app themes to choose from. It will be fairly easy to submit your own themes in the future, I hope, since the MM is no longer themable via Substratum. (_fact check me if I'm wrong_)

---

Also _it has literally nothing to do with kotlin_ 😏

**Comment by xerta555** [comment]
> In the redesign we'll have in-app themes to choose from. It will be fairly easy to submit your own themes in the future, I hope, since the MM is no longer themable via Substratum. (fact check me if I'm wrong)

Very good initiative, although personally I don't use any substratum themes for Messrs.
Can you tell me up to what level of personalization it will be possible to go?

**Comment by diareuse** [comment]
From user's perspective:
User can select predefined theme.

From developer's perspective:
[Any widget, any color can be themed](https://github.com/diareuse/Magisk/blob/feature/redesign/app/src/main/res/values/styles_md2.xml)

**Comment by xerta555** [comment]
> From developer's perspective:
Any widget, any color can be themed

I was thinking more of a color editor like Telegram's that allows you to change the color of an UI elements.

Would it be possible to create your own personalized theme that you can modify on the fly?

**Comment by diareuse** [comment]
No, that will not be possible.

**Comment by xerta555** [comment]
> No, that will not be possible.

Okay, so if I understand correctly, maximum customization of the interface will only be possible through substratum themes?

**Comment by diareuse** [comment]
No, Substratum theming is no longer possible. The newest versions of MM are kind of protected against that.
The only option will be in-app themes defined by **me** or other developers.
---
Since John likes to dig deep into the app, it has to be as low maintenance as possible. Having fully user-definable resources would defy that.

**Comment by xerta555** [comment]
>  The only option will be in-app themes defined by me or other developers.

Ok so i guess there will be for example the possibility to define a dark material theme and change the color of all texts with a color like Blue or Orange for example?

**Comment by xerta555** [comment]
> Yes. I basically allow to theme any color from primary, secondary to background and everything in between.

> The first version will ship with one default theme, which has both light and dark variant. More will come.

It will be beautiful if for black material theme we could get a black, a grey material theme and one more for AMOLED devices in addition as u mean of a white theme as actually, but i think that u already have planned to add them by default.

**Comment by diareuse** [comment]
Yeah, well... Having two variants of black themes implies having two sets of themes instead of one. So there will be `Not white` and `Not black` themes. 

This is the state we're at right now:
![merged](https://user-images.githubusercontent.com/5982859/67028745-2b4ed900-f10c-11e9-9009-83243275626e.png)

The dark theme needs some text brightening though.

---

I hope this suffices your expectations, gotta get back to work.

**Comment by xerta555** [comment]
> This is the state we're at right now:

The appearance of the interface in light and dark really changes from the current appearance on Canary, Beta and Stable channels, this remains a very good design!

> The dark theme needs some text brightening though.

Yes, i imagine from a readability point of view for the user this is undeniable.

> I hope this suffices your expectations, gotta get back to work.

As it stands, yes, i now clearly have a good idea of what will be possible with the next versions, looking forward to testing it!
